parsing/parser.mly

```diff
@@ -2130,6 +2105,30 @@ let_pattern:
       { $1 }
 ;
 
+%inline array_set(dot, left, mid, right):
+  | simple_expr dot left mid right LESSMINUS expr
+      { $1, $2, $3, $4, Some $7 }
```

Comment:
I would prefer to use named parameters instead of positions, especially for `$7`.

I think it is weird that there are four parameters that are always used in the same way in the grammar. Could they be packed into a single symbol that returns a tuple?

Reply:
The difficulty is that the other function in the cartesian product would need to produce the symbol variants, rather than use there sub symbol directly. There might be a better characterization, but it doesn't seem that straightforward.